Src/FluentAssertions/Execution/AssertionScope.cs

```diff
@@ -69,15 +85,33 @@ public AssertionScope(string context)
         /// </summary>
         public string Context { get; set; }
 
+#if NOTNET45
         /// <summary>
         /// Gets the current thread-specific assertion scope.
         /// </summary>
         public static AssertionScope Current
         {
-            get => current ?? new AssertionScope(new DefaultAssertionStrategy());
-            private set => current = value;
+
+            get => current.Value ?? new AssertionScope(new DefaultAssertionStrategy());
+            private set => current.Value = value;
+
         }
+#else
+        /// <summary>
+        /// Gets the current thread-specific assertion scope.
+        /// </summary>
+        public static AssertionScope Current => LogicalCurrent(); 
 
+        public static AssertionScope LogicalCurrent()
+        {
+            if (CallContext.LogicalGetData("this") == null)
+            {
+                CallContext.LogicalSetData("this", new AssertionScope(new DefaultAssertionStrategy()));
```

Comment:
❓ Currently when `current` is `null` a new `AssertionScope` is constructed and returned, but not written to `current`.
Why the difference here?

Reply:
Fix